### PR TITLE
Support blocking operators in OperatorAssertion [spilljoin]

### PR DESCRIPTION
**Pull request by findepi**

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/operator/OperatorAssertion.java [hunk 1]
```diff
@@ -34,8 +34,6 @@
 import static com.facebook.presto.type.TypeJsonUtils.appendToBlockBuilder;
 import static com.google.common.base.Preconditions.checkArgument;
 import static io.airlift.testing.Assertions.assertEqualsIgnoreOrder;
-import static org.testng.Assert.assertFalse;
-import static org.testng.Assert.assertTrue;
 
 public final class OperatorAssertion
```
Comment (findepi) [inline presto-main/src/test/java/com/facebook/presto/operator/OperatorAssertion.java]: Dear Reviewer,

The "Refactor OperatorAssertion.toPages" commit should not be read as a diff. The functions have been rewritten.

File: presto-main/src/test/java/com/facebook/presto/operator/OperatorAssertion.java [hunk 2]
```diff
@@ -192,15 +197,14 @@ public static void assertOperatorEqualsIgnoreOrder(
...
-        for (int hashChannel : channels) {
-            types.remove(hashChannel - removed);
-            removed++;
-        }
-        return ImmutableList.copyOf(types);
+        Set<Integer> indexesSet = ImmutableSet.copyOf(indexes);
+
+        return IntStream.range(0, list.size())
+                .filter(index -> !indexesSet.contains(index))
+                .mapToObj(ImmutableList.copyOf(list)::get)
```
Comment (losipiuk) [inline presto-main/src/test/java/com/facebook/presto/operator/OperatorAssertion.java]: Drop this `ImmutableList.copyOf` wrapper. It reduces clarity and we do expect random-access lists here anyway.

File: presto-main/src/test/java/com/facebook/presto/operator/TestingOperatorContext.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                1,
+                new PlanNodeId("test"),
+                "operator type"
+        );
+
+        return operatorContext;
+    }
+
+    private TestingOperatorContext()
+    {
```
Comment (losipiuk) [inline presto-main/src/test/java/com/facebook/presto/operator/TestingOperatorContext.java]: put `{}` in line with method declaration

**Comment by findepi** [comment]
@losipiuk comments applied

**Comment by losipiuk** [comment]
looks good - please squash the fixups

**Comment by findepi** [comment]
squashed. Thanks for review!

**Comment by losipiuk** [comment]
Merged. Thanks!

**Pull request closed without merging**